payload.go

```diff
@@ -36,12 +38,18 @@ func NewPayload(buf []byte) (*Payload, error) {
 		return nil, errors.New("bad payload (priority should repeat)")
 	}
 
-	nicID, err := UUIDToStr(uuidbuf)
+	nicID, err := UUIDToStr(buf[8:24])
+	if err != nil {
+		return nil, err
+	}
+
+	msgID, err := UUIDToStr(buf[24:40])
```

Comment:
It will return an error if receiving a message from an older exoip since the payload  isn't big enough 24 bits vs 40 with the newer version. You should check for the *previousVersion* and set a `none` for the uuid for example.

Reply:
the allocated buffer will be long enough. But I agree, it's not obvious reading this section of code.

Reply:
Ok, but the UUIDToStr() will return an error if it cannot parse the content, which obviously is going to be empty with an older message. A test would be great to validate the parsing of an old version message.

Reply:
old versions are shorter, uuid-to-str does formatting and will display bunch of zeros. the old version not being future compatible will however complain a lot.